proto/ui.proto

```diff
@@ -0,0 +1,106 @@
+syntax = "proto3";
+
+package ui;
+
+import "api.proto";
+
+option go_package = "diagonal.works/b6/proto";
+
+message UIRequestProto {
+    api.NodeProto node = 1;
+    string expression = 2;
+}
+
+message UIResponseProto {
+    StackProto stack = 1;
+    api.NodeProto node = 2;
+    FeatureIDsProto highlighted = 3;
+    repeated string queryLayers = 4;
+}
+
+message StackProto {
+    repeated SubstackProto substacks = 1;
+}
+
+message SubstackProto {
+    repeated LineProto lines = 1;
+    bool collapsable = 2;
+}
+
+message LineProto {
+    oneof line {
+        ValueLineProto value = 1;
+        ValuePairLineProto valuePair = 2;
+        ExpressionLineProto expression = 3;
+        TagsLineProto tags = 4;
+        HistogramBarLineProto histogramBar = 5;
+        ShellLineProto shell = 6;
+        ErrorLineProto error = 7;
+    }
+}
+
+message ValueLineProto {
+    AtomProto atom = 1;
+    api.NodeProto clickExpression = 2;
+}
+
+message ValuePairLineProto {
+    AtomProto first = 1;
```

Comment:
this feels a bit awkward, u didnt wanna 
message ValuePairLineProto {
   ValueLineProto first = 1
   ValueLineProto second = 2
 }  ?

Reply:
Yes, it was awkward - have improved. I added a new proto, ClickableAtom proto, since Atoms aren't generally clickable, and also conceptually a ValuePairLine doesn't actually embed ValueLines (since they have borders etc).